File: common/src/main/java/com/ais/middleware/common/events/policy/IssuanceSagaStartedEvent.java
package com.ais.middleware.common.events.policy;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.OffsetDateTime;

/**
 * Published by PolicyIssuanceAndLifecycleManagement immediately when the IssuanceSaga is created.
 * Downstream systems should not react to this event because it exists only as an audit marker.
 */
public record IssuanceSagaStartedEvent(
        String issuanceId,
        String accountId,
        String submittingChannel,
        OffsetDateTime startedAt
) {
    @JsonCreator
    public IssuanceSagaStartedEvent(
            @JsonProperty("issuanceId") String issuanceId,
            @JsonProperty("accountId") String accountId,
            @JsonProperty("submittingChannel") String submittingChannel,
            @JsonProperty("startedAt") OffsetDateTime startedAt
    ) {
        this.issuanceId = issuanceId;
        this.accountId = accountId;
        this.submittingChannel = submittingChannel;
        this.startedAt = startedAt;
    }
}

File: dotnet/Middleware.Contracts/Events/IssuanceSagaStartedEvent.cs
using NServiceBus;

namespace Middleware.Contracts.Events;

public sealed class IssuanceSagaStartedEvent : IEvent
{
    public string IssuanceId { get; set; } = string.Empty;
    public string AccountId { get; set; } = string.Empty;
    public DateTimeOffset StartedAt { get; set; } = DateTimeOffset.UtcNow;
}
